examples/client_tcp/Client_main.c

```diff
@@ -0,0 +1,186 @@
+#define SOCKET_PORT                 5000
+#define TCP_DATA_RCV_WITHOUT_FLAGS  0
+#define RCV_BUFFER_SIZE             520
+#define SEND_BUFFER_SIZE            500 
+
+/****************************************************************************
+ * Client_tcp_main
+ ****************************************************************************/
+
+int main(int argc, char *argv[])
+{
+  int socket_client = 0;
+  char rcv_buffer[RCV_BUFFER_SIZE];
+  char buffer_to_send[SEND_BUFFER_SIZE];
+  int bytes_read_from_server = 0;
+  struct sockaddr_in serv_addr;
+  protocolo_ipc tprotocol;
+
+  /* Check if there are sufficient arguments passed to this program */
+
+  if (argc != 2)
+    {
+      printf("\nNot enough parameters: %s.\n", argv[0]);
+      return 1;
+    }
+
+  /* Create client socket */
+
+  memset(rcv_buffer, 0x00, sizeof(rcv_buffer));
+  socket_client = socket(AF_INET, SOCK_STREAM, 0);
+  if (socket_client < 0)
+    {
+      perror("Failed to create client socket");
+      exit(EXIT_FAILURE);
+    }
+
+  /* Connect to server socket */
+
+  memset(&serv_addr, 0x00, sizeof(serv_addr));
+  serv_addr.sin_family = AF_INET;
+  serv_addr.sin_port = htons(SOCKET_PORT);
+
+  if (inet_pton(AF_INET, argv[1], &serv_addr.sin_addr) <= 0)
+    {
+      perror("Failed when executing inet_pton()");
+      exit(EXIT_FAILURE);
+    }
+
+  if (connect(socket_client, (struct sockaddr *)&serv_addr,
+              sizeof(serv_addr)) < 0)
+    {
+      perror("Failed to connect to server socket");
+      exit(EXIT_FAILURE);
+    }
+
+  /* Countinuosly send server data to be forwarded to LPWAN transceiver */
+
+  while (1)
+    {
+      /* Formats message to be sent to server (opcode, message size
+       * and message content)
+       */
+
+      tprotocol.opcode = 'U';
+      tprotocol.msg_size = 4;
+      snprintf((char *)tprotocol.msg, sizeof(tprotocol.msg), "0102");
+
+      /* Send message to server */
+
+      memcpy(buffer_to_send, (unsigned char *)&tprotocol,
+             sizeof(protocolo_ipc));
+      write(socket_client, buffer_to_send, strlen(buffer_to_send));
+      printf("Message sent to server!\n\n");
+
+      /* Waits for server response */
+
+      bytes_read_from_server = recv(socket_client, rcv_buffer,
+                               sizeof(protocolo_ipc),
+                               TCP_DATA_RCV_WITHOUT_FLAGS);
+
+      if (bytes_read_from_server < 0)
+        {
+          perror("Failed to get server response");
+          exit(EXIT_FAILURE);
+        }
+      else
+        {
+          /* Server response successfully received. Print it on the screen */
+
+          memcpy((unsigned char *)&tprotocol, rcv_buffer,
+                 sizeof(protocolo_ipc));
+          printf("Protocol: opcode: %c\n", tprotocol.opcode);
+          printf("Protocol: message size: %d\n", tprotocol.msg_size);
+          printf("Protocol: message: %s\n", tprotocol.msg);
+        }
+
+      /* Wait 15 seconds to send again */
+
+      sleep(15);
```

Comment:
Please use a macro to define this value, to make it easier to user change it in the top of this file.